src/context.h

```diff
@@ -203,6 +203,15 @@ struct ly_ctx;
                                         'require-instance false;'. It also enables usage of
                                         [lyd_leafref_get_links](@ref lyd_leafref_get_links) and
                                         [lyd_leafref_link_node_tree](@ref lyd_leafref_link_node_tree) APIs. */
+#define LY_CTX_STORE_INVALID_DATA 0x0800 /**< By default, storing of value of given data node is subject to build-in validations
+                                        as patterns, length, ranges etc.. This option bypass these validations, and allow user
+                                        to validate the values manually by additional calls as lyd_validate.
+                                        It also supresses usage of advanced built-in plugin types (ipv4-address, ipv6-address etc..),
+                                        which expects that value is already valid during storage operation. Instead the value is
+                                        processed by basic plungin (usually string).
+                                        Change of this flag during the lifetime of context is not fully supported, once the advanced
+                                        plugins are loaded during context creation, they will be used regardless of this flag.
+                                        Therefore it is recommended to set this flag during the context creation */
```

Comment:
Unless I'm very wrong, these plugins implement not just data storage, but also conversion of values to their canonical forms. Many types are defined to accept a wide range of textual inputs where many "different" inputs will result in a single canonical value. Examples are uppercase vs. lowercase values in `ietf-yang-types:hex-string`. If you disable these plugins, values will not be converted to their canonical form. I suspect that this will in turn break any reasonable `when` and `must` statements because these are defined to use comparisons against the canonical form.

If this patch gets accepted, at least the documentation should be clear about these implications.

Reply:
True, but in my case, it is acceptable. Description was extended with such consequence.